src/core/TriangleSplitter.js

```diff
@@ -189,7 +221,28 @@ export class TriangleSplitter {
 				// so we can use that information to determine whether to split later.
 				const startDist = plane.distanceToPoint( _edge.start );
 				const endDist = plane.distanceToPoint( _edge.end );
-				if ( Math.abs( startDist ) < COPLANAR_EPSILON && Math.abs( endDist ) < COPLANAR_EPSILON ) {
+				let coPlanarEpsilonStart = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.start.distanceTo( planeCenter ) / planeEdgeSize );
+				let coPlanarEpsilonEnd = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.end.distanceTo( planeCenter ) / planeEdgeSize );
```

Comment:
Do you mind explaining the rationale behind these calculations? Including the thinking behind "planeCenter"? It seems as though the distance to an arbitrarily chosen point on the plane will have a significant impact.

Reply:
We don't want to use an epsilon larger than COPLANAR_EPSILON, that's why we use the `Math.max`
Please look at the picture posted with https://github.com/gkjohnson/three-bvh-csg/issues/199#issuecomment-1986287165 for the annotations:
We scale epsilon (`e`) to get epsilon scaled (`E`)  by IP / AB.
  * AB is the min edge size
  * instead of using `IM` like in my draft  we use `IE` or `IF`, with `E` and `F` start and end points of the edge `_edge`
 This computation is from the Thales theorem.

Reply:
Thanks I've taken a look at the diagrams and I understand the intent. Basically a very small edge leads to more uncertainty in the true plane position so you want to use a large epsilon when checking distances to such a plane. And you're effectively scaling the amount that we scale the epsilon by the distance to the point used to generate the plane since we know that point is exactly correct.

This may be a bit difficult to describe effectively but it would nice to have some comments in the code to describe what's happening here and why. Even if it includes a link to this comment thread for context.

Reply:
Hi @gkjohnson I have added some comments and a link to the Github issue in the last commit